### PR TITLE
Add deck to papermill plugin task

### PR DESCRIPTION
**Pull request by CalvinLeather**
# TL;DR
This PR modifies the papermill plugin so that, if render_deck=True is passed to __init__, the rendered output html is inserted as a flyte deck.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [x] Any pending items have an associated Issue

## Complete description
This PR modifies the papermill plugin so that, if a kwarg render_deck is passed, the flyte deck is added automatically, without need for a follow up task. The render_deck kwarg was used to ensure backwards compatibility, by default, the behaviour of the papermill plugin won't be changed (i.e., this deck support is opt-in).

Note that this won't happen for cached execution, there is an open issue out for fixing flyte decks for cached execution.

I tested this change by installing this branch of flytekit into a container and registering one of our company's test notebook tasks, and running it. It gets a deck as intended:
image

image

Not 100% happy with how I tested this plugin. Since the unit tests use local execution, the deck/context functionality I think isn't getting used. Let me know if there is a better way to do test this.

## Tracking Issue
n/a (can open an issue if you like, JFDI'd this after discussing with Ketan [here](https://flyte-org.slack.com/archives/CP2HDHKE1/p1658958351388009?thread_ts=1658863230.650909&cid=CP2HDHKE1))

## Follow-up issue
_NA_

### CODE REVIEW

**Comment by CalvinLeather** [comment]
Did a quick force push to my branch to --signoff in a rebase.

**Review by pingsutw — APPROVED** [review]
This is awesome!! Thank you @CalvinLeather. LGTM
